xla/mlir/tools/mlir_interpreter/dialects/comparators.h

```diff
@@ -43,14 +44,21 @@ struct FloatCompare : CwiseAll {
     }
   }
 
+  // Template type for non-floating point and non-complex
   template <typename T>
   static bool isnan(T a) {
-    return std::isnan(a);
+    return false;
   }
+  // Handling complex types
   template <typename T>
   static bool isnan(std::complex<T> a) {
     return std::isnan(std::real(a)) || std::isnan(std::imag(a));
   }
+  // Template for isnan (Floating numbers)
+  template <typename T>
+  static bool isnan(std::is_floating_point<T> a) {
+    return std::isnan(a);
+  }
```

Comment:
What motivates this change?

Reply:
Hi @beckerhe xla-mlir tests(//xla/mlir/tools/mlir_interpreter/dialects/tests/...) are failing on the Windows with the error below. The detailed error log is atta
[xla-mlir-failure.txt](https://github.com/user-attachments/files/17455936/xla-mlir-failure.txt)
ched
C:\Program Files (x86)\Windows Kits\10\include\10.0.22621.0\ucrt\corecrt_math.h(419,16): error: call to 'fpclassify' is ambiguous
  419 |         return fpclassify(_X) == FP_NAN;
      |                ^~~~~~~~~~
.\xla/mlir/tools/mlir_interpreter/dialects/comparators.h(48,17): note: in instantiation of function template specialization 'isnan<signed char>' requested here
   48 |     return std::isnan(a);
      |                 ^
.\xla/mlir/tools/mlir_interpreter/dialects/comparators.h(34,9): note: in instantiation of function template specialization 'mlir::interpreter::FloatCompare<0, true, false>::isnan<signed char>' requested here
   34 |     if (isnan(a) || isnan(b)) return nan_result;
      |         ^
.\xla/mlir/tools/mlir_interpreter/framework/interpreter_value_util.h(114,19): note: in instantiation of function template specialization 'mlir::interpreter::FloatCompare<0, true, false>::Apply<signed char>' requested here
  114 |       return {Fn::Apply(t, std::get<T>(rhs.storage))};
      |                   ^

Reply:
Thank you for the full error message. I believe you're hitting a bug in MSVC.

According to the release notes (https://learn.microsoft.com/en-us/visualstudio/releases/2022/release-notes) it has been fixed in version 17.11 of Visual Studio 2022. (`Missing integral types overload for cmath functions` in top bug fixes)

Is it feasible for you to update MSVC?

If not we can certainly change the code, but I'm reluctant since the code is actually correct in its current form.

Reply:
Hi @beckerhe, yes, I tested on my end, 17.11 works well. Thank you. I will make the suggested changes asap.